mood-analysis.rb

```diff
@@ -1,32 +1,76 @@
 FEELINGS = {
-  happy: %w(yay, good, great),
-  sad: %w(terrible, awful, horrible)
+  happy: %w(yay good great),
+  sad: %w(terrible awful horrible)
 }
 
+@text = [
+  "03/01 I'm having a terrible horrible no good day.",
+  "03/13 Yesterday was horrible, but today is great! Yay!",
+  "04/02 Sad Panda. #terribleday",
+  "04/15 Hello World, today is fabulous! #yay",
+  "05/01 Great! Yay! Good! Yay! Happy. Happy.",
+  "05/11 Yay, yay, yay! I'm having a awfuly great day."
+]
+
+def strip_punctuation(words)
+  return words.downcase.gsub(/[^a-z0-9\s]/i, '')
+end
+
 def analyze_mood(words)
   happy = 0
   sad = 0
   words.downcase!
-  words.split(" ").each do |word|
+  words_stripped = strip_punctuation(words)
+  words_stripped.split(" ").each do |word|
     if FEELINGS[:happy].include? word
       happy += 1
     elsif FEELINGS[:sad].include? word
       sad += 1
     end
   end
+
   return ":-)" if happy > sad
   return ":-(" if happy < sad
   return ":-|"
 end
 
-text = [
-  "03/01 I'm having a terrible horrible no good day.",
-  "03/13 Yesterday was horrible, but today is great! Yay!",
-  "04/02 Sad Panda. #terribleday",
-  "04/15 Hello World, today is fabulous! #yay",
-  "05/01 Great! Yay! Good! Yay! Happy. Happy.",
-  "05/11 Yay, yay, yay! I'm having a awfuly great day."
-]
+def print_moods
+  @text.each do |each_entry|
+    puts each_entry[0..4] + " " + analyze_mood(each_entry)
+  end
+end
+
+def happy_days
+  desired_happy = 3
+  count_happy = 0
+  entry_count = 0  # is count automated somewhere else?
+
+  @text.each do |each_entry|
+    entry_count += 1
+    mood = analyze_mood(each_entry)
+    if mood == ":-)"
+      count_happy += 1
+    end
+    if count_happy == 3
+      return puts "It takes #{entry_count} entries for 3 happy days to occur."   # Is it correct to return a puts since we just want a console output??
+    end
+  end
+  return puts "We did not find 3 happy days in the input data."
+end
+
+def overall_mood
+  happy_count = 0
+  sad_count = 0
+
+  @text.each do |each_entry|
+    mood = analyze_mood(each_entry)
+    happy_count += 1 if mood == ":-)"
+    sad_count += 1 if mood == ":-("
+  end
```

Comment:
What about :-|? What if there more moods? 

This could be generalized. Also I wanted you all to recognize that using a hash here is ideal. In fact, hashes are used quite often in this case (counting an undetermined number of things), especially in whiteboarding problems.

Here's my solution as reference:
```
def overall_mood(entries)
  return nil if entries.length == 0
  emoticons = Hash.new(0)
  entries.each do |entry|
    emoticon = analyze_mood(entry)
    emoticons[emoticon] += 1
  end
  return emoticons.max_by{|k,v| v}[0]
end
```